src/charm.py

```diff
@@ -189,22 +200,76 @@ def _on_leader_elected(self, _) -> None:
                 )
         self.unit_peer_data.update({"leader": "true"})
 
+        # Create and set cluster and cluster-set names in the peer relation databag
+        common_hash = generate_random_hash()
+        self.app_peer_data.setdefault(
+            "cluster-name", self.config.cluster_name or f"cluster-{common_hash}"
+        )
+        self.app_peer_data.setdefault("cluster-set-domain-name", f"cluster-set-{common_hash}")
```

Comment:
Q: will we overwrite `cluster-name` and `cluster-set-domain-name` on the leader move (re-elected)?